llvm/lib/Transforms/Scalar/Float2Int.cpp

```diff
@@ -382,9 +382,18 @@ bool Float2IntPass::validateAndTransform() {
       continue;
     }
 
-    // OK, R is known to be representable. Now pick a type for it.
-    // FIXME: Pick the smallest legal type that will fit.
-    Type *Ty = (MinBW > 32) ? Type::getInt64Ty(*Ctx) : Type::getInt32Ty(*Ctx);
+    // OK, R is known to be representable.
+    // Pick the smallest legal type that will fit.
+    Type *Ty;
+    if (MinBW <= 8) {
+      Ty = Type::getInt8Ty(*Ctx);
+    } else if (MinBW <= 16) {
+      Ty = Type::getInt16Ty(*Ctx);
+    } else if (MinBW <= 32) {
+      Ty = Type::getInt32Ty(*Ctx);
+    } else {
+      Ty = Type::getInt64Ty(*Ctx);
+    }
```

Comment:
This needs to be based on the legal integer types provided by DataLayout.

Reply:
@nikic Fixed!

Reply:
> This needs to be based on the legal integer types provided by DataLayout.

Well, every time I rerun the tests it seems that 64 and 32 are not legal integer types for some reason. I have no clue

Reply:
You'll have to adjust the tests to specify a suitable data layout, see https://llvm.org/docs/LangRef.html#data-layout.